pimd/pim_sock.c

```diff
@@ -154,61 +137,61 @@ int pim_socket_mcast(int protocol, struct in_addr ifaddr, struct interface *ifp,
 		close(fd);
 		return PIM_SOCK_ERR_DSTADDR;
 #endif
-	}
 
 
-	/* Set router alert (RFC 2113) for all IGMP messages (RFC 3376 4.
-	 * Message Formats)*/
-	if (protocol == IPPROTO_IGMP) {
-		uint8_t ra[4];
-		ra[0] = 148;
-		ra[1] = 4;
-		ra[2] = 0;
-		ra[3] = 0;
-		if (setsockopt(fd, IPPROTO_IP, IP_OPTIONS, ra, 4)) {
-			zlog_warn(
-				"Could not set Router Alert Option on socket fd=%d: errno=%d: %s",
-				fd, errno, safe_strerror(errno));
-			close(fd);
-			return PIM_SOCK_ERR_RA;
+#if PIM_IPV == 4
+		/* Set router alert (RFC 2113) for all IGMP messages (RFC
+		 * 3376 4. Message Formats)*/
+		if (protocol == IPPROTO_IGMP) {
+			uint8_t ra[4];
+			ra[0] = 148;
+			ra[1] = 4;
+			ra[2] = 0;
+			ra[3] = 0;
+			if (setsockopt(fd, IPPROTO_IP, IP_OPTIONS, ra, 4)) {
+				zlog_warn(
+					"Could not set Router Alert Option on socket fd=%d: errno=%d: %s",
+					fd, errno, safe_strerror(errno));
+				close(fd);
+				return PIM_SOCK_ERR_RA;
+			}
 		}
-	}
+#endif
 
-	{
-		int reuse = 1;
-		if (setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, (void *)&reuse,
-			       sizeof(reuse))) {
-			zlog_warn(
-				"Could not set Reuse Address Option on socket fd=%d: errno=%d: %s",
-				fd, errno, safe_strerror(errno));
-			close(fd);
-			return PIM_SOCK_ERR_REUSE;
-		}
-	}
+		sockopt_reuseaddr(fd);
 
-	{
-		const int MTTL = 1;
-		int ttl = MTTL;
-		if (setsockopt(fd, IPPROTO_IP, IP_MULTICAST_TTL, (void *)&ttl,
+		if (setsockopt(fd, PIM_AF_OPT, PIM_AF_MCAST_TTL, (void *)&ttl,
 			       sizeof(ttl))) {
 			zlog_warn(
 				"Could not set multicast TTL=%d on socket fd=%d: errno=%d: %s",
 				MTTL, fd, errno, safe_strerror(errno));
 			close(fd);
 			return PIM_SOCK_ERR_TTL;
 		}
-	}
 
-	if (setsockopt_ipv4_multicast_loop(fd, loop)) {
-		zlog_warn(
-			"Could not %s Multicast Loopback Option on socket fd=%d: errno=%d: %s",
-			loop ? "enable" : "disable", fd, errno,
-			safe_strerror(errno));
-		close(fd);
-		return PIM_SOCK_ERR_LOOP;
-	}
+#if PIM_IPV == 4
+		if (setsockopt_ipv4_multicast_loop(fd, loop)) {
```

Comment:
wrong indentation

`#if` does not add a level of indent

Reply:
Ack